service/core/server/src/main/scala/com/lightbend/lagom/internal/server/ServiceRouter.scala

```diff
@@ -183,13 +187,15 @@ private[lagom] abstract class ServiceRouter(httpConfiguration: HttpConfiguration
       handleServiceCall(serviceCall, descriptor, requestSerializer, responseSerializer, filteredHeaders, request)
         .recover {
           case NonFatal(e) =>
-            logException(e, descriptor, call)
-            exceptionToResult(descriptor, filteredHeaders, e)
+//            logException(e, descriptor, call)
+            val result = processError(request, e)
+            Await.result(result, Duration.Inf)
```

Comment:
There shouldn't be an `Await` here, as this will cause the thread to block. I believe the right way to do this would be to to replace `recover` above with `recoverWith` and return `result`.